lib/config/hadeployconfigverify.go

```diff
@@ -0,0 +1,271 @@
+	errorList := list.New()
+
+	// Check if config file path is empty
+	if len(configFile) < 1 {
+		return errors.New("config file path is empty")
+	}
+
+	// Parse the config file
+	haDeployConfig, err := c.Parse(configFile)
+	if err != nil {
+		errorList.PushBack(err.Error())
+	} else {
+		c = haDeployConfig
+	}
+
+	if haDeployConfig == nil || haDeployConfig.Architecture == nil {
+		return errors.New("invalid ha deploy config")
+	}
+
+	// Validate ExistingInfra (on prem)
+	if haDeployConfig.Architecture.ExistingInfra != nil {
+		c.verifyConfigInitials(haDeployConfig.Architecture.ExistingInfra, errorList)
+		c.validateExistingInfraBackupConfig(errorList)
+		c.verifyExistingInfraSettings(haDeployConfig.ExistingInfra.Config, errorList)
+
+		// on prem aws or self-managed
+		if isExternalDb(haDeployConfig) {
+			c.verifyExternalPgSettings(haDeployConfig.External.Database.PostgreSQL, errorList)
+			c.verifyExternalOsSettings(haDeployConfig.External.Database.OpenSearch, errorList)
+			c.verifyAwsExternalOsSettings(haDeployConfig.External.Database.OpenSearch.Aws, errorList)
+		}
+	}
+
+	// Validate Aws
+	if haDeployConfig.Architecture.Aws != nil {
+		c.verifyConfigInitials(haDeployConfig.Architecture.Aws, errorList)
+		c.validateAwsBackupConfig(errorList)
+		c.validateAwsSettings(haDeployConfig.Aws.Config, errorList)
+	}
+
+	// Validate common fields
+	c.verifyAutomateSettings(errorList)
+	c.verifyChefServerSettings(errorList)
+	c.verifyOpensearchSettings(errorList)
+	c.verifyPostgresqlSettings(errorList)
+
+	return getSingleErrorFromList(errorList)
+
+}
+
+func (c *HaDeployConfig) verifyConfigInitials(configInitials *ConfigInitials, errorList *list.List) {
+	validateRequiredStringTypeField(configInitials.SecretsKeyFile, "secrets_key_file", errorList)
+	validateRequiredStringTypeField(configInitials.SecretsStoreFile, "secrets_store_file", errorList)
+	validateRequiredStringTypeField(configInitials.Architecture, "Architecture", errorList)
+	validateRequiredStringTypeField(configInitials.WorkspacePath, "workspace_path", errorList)
+	validateRequiredStringTypeField(configInitials.SSHUser, "ssh_user", errorList)
+	validateRequiredPathField(configInitials.SSHKeyFile, "ssh_key_file", errorList)
+	validateStringTypeField(configInitials.SSHGroupName, "ssh_group_name", errorList)
+	validateStringTypeField(configInitials.LoggingMonitoringManagement, "logging_monitoring_management", errorList)
+	validateStringTypeField(configInitials.HabitatUIDGid, "habitat_uid_gid", errorList)
+	validateBackupMount(configInitials.BackupMount, errorList)
+}
+
+func (c *HaDeployConfig) validateExistingInfraBackupConfig(errorList *list.List) {
+	// validate existing infra backup config
+	if c.Architecture.ExistingInfra.BackupConfig == "object_storage" {
+		c.verifyObjectStorage(c.ObjectStorage.Config, errorList)
+	} else if c.Architecture.ExistingInfra.BackupConfig == "file_system" {
+		// no check needed
+	} else {
+		errorList.PushBack("Invalid or empty backup_config")
```

Comment:
Put this as the first condition and return

Reply:
backupConfig := c.Architecture.ExistingInfra.BackupConfig
	if backupConfig != "object_storage" || backupConfig != "file_system" {
		errorList.PushBack("Invalid or empty backup_config")
		return
	}
	if backupConfig == "object_storage" {
		c.verifyObjectStorage(c.ObjectStorage.Config, errorList)
	}
	
	Modified to this